hikari/impl/shard.py

```diff
@@ -430,31 +435,38 @@ async def request_guild_members(
         include_presences: undefined.UndefinedOr[bool] = undefined.UNDEFINED,
         query: str = "",
         limit: int = 0,
-        user_ids: undefined.UndefinedOr[typing.Sequence[snowflakes.SnowflakeishOr[users.User]]] = undefined.UNDEFINED,
+        users: undefined.UndefinedOr[typing.Sequence[snowflakes.SnowflakeishOr[users_.User]]] = undefined.UNDEFINED,
         nonce: undefined.UndefinedOr[str] = undefined.UNDEFINED,
     ) -> None:
+        await self._chunk_request_ratelimiter.acquire()
+
```

Comment:
If we raise an exception on line 445, do we need to unacquire this? Should it go after the validation checks to prevent acquiring time and not using it, maybe?

Reply:
Do we check the nonce size if present?

sending bad data will kick the gateway connection

Reply:
if the nonce is over 32 byte characters long Discord will just ignore it, not kick the gateway connection

Reply:
I've moved the acquire after the validation, and added a bytes length check for nonce, if there's not anything else for this comment I'm missing then you can mark this as resolved